src/main/java/Duke.java

```diff
@@ -1,10 +1,103 @@
+import java.util.Scanner;
+
 public class Duke {
+    public static final String INITIALISING = "initialising";
+    public static final int MAX_LIST_SIZE = 100;
+    private static Task[] list = new Task[MAX_LIST_SIZE];
+    private static int listSize = 0;
+
+    public static void addToList(Task task) {
+        list[listSize++] = task;
+        System.out.println("Okay! I have added this:");
+        System.out.printf("\t%s\n", task);
+        System.out.printf("Now you have %d task%s in the list.\n", getListSize(),
+                (getListSize() == 1)? "": "s");
+    }
+
+    public static int getListSize() {
+        return listSize;
+    }
+
+    public static void displayList() {
+        System.out.println("Here's your TODO");
+        for (int i = 0; i < listSize; i++) {
+            System.out.printf("%d.%s\n", i + 1,list[i]);
+        }
+    }
+
+    public static void markTaskAsDone(String task) {
+        int taskNumber = Integer.parseInt(task);
+        list[taskNumber - 1].setDone();
+        System.out.println("I have marked your task as done!");
+        System.out.printf("\t%s\n", list[taskNumber - 1]);
+    }
+
+    public static void updateList(String line) {
+        int dividerPosition = line.indexOf(" ");
+        String command = line.substring(0, dividerPosition);
+        String task = line.substring(dividerPosition + 1);
+
+        if (command.equals("done")) {
+            markTaskAsDone(task);
+
+        } else if (command.equals("todo")) {
+            addToList(new Todo(task));
+
+        } else if (command.equals("deadline")) {
+            int byPosition = line.indexOf("/by");
```

Comment:
Consider commenting on the usage of numbers such as 1 and 4 to explain how it affects the indexing.